svm/programs/executor-quoter/src/instructions/get_quote.rs

```diff
@@ -0,0 +1,249 @@
+        chain_info.native_decimals,
+        gas_limit,
+        msg_value,
+    )?;
+
+    // Return the quote as big-endian U256 (32 bytes) via set_return_data.
+    // Clients can read this via simulateTransaction or CPI callers via get_return_data.
+    set_return_data(&required_payment.to_be_bytes());
+
+    Ok(())
+}
+
+/// Process RequestExecutionQuote instruction.
+/// Returns the required payment, payee address, and quote body.
+///
+/// Accounts: Same as RequestQuote
+pub fn process_request_execution_quote(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    data: &[u8],
+) -> ProgramResult {
+    // Parse accounts
+    let [config_account, chain_info_account, quote_body_account] = accounts else {
+        return Err(ProgramError::NotEnoughAccountKeys);
+    };
+
+    // Load accounts (discriminator checked inside load_account)
+    let config = load_account::<Config>(config_account, program_id)?;
+
+    let chain_info = load_account::<ChainInfo>(chain_info_account, program_id)?;
+    if !chain_info.is_enabled() {
+        return Err(ExecutorQuoterError::ChainDisabled.into());
+    }
+
+    let quote_body = load_account::<QuoteBody>(quote_body_account, program_id)?;
+
+    // Parse instruction data to get relay_instructions
+    if data.len() < 70 {
+        return Err(ExecutorQuoterError::InvalidInstructionData.into());
+    }
+
+    let mut len_bytes = [0u8; 4];
+    len_bytes.copy_from_slice(&data[66..70]);
+    let request_bytes_len = u32::from_le_bytes(len_bytes) as usize;
+    let relay_start = 70 + request_bytes_len;
+
+    if data.len() < relay_start + 4 {
+        return Err(ExecutorQuoterError::InvalidInstructionData.into());
+    }
+
+    let mut relay_len_bytes = [0u8; 4];
+    relay_len_bytes.copy_from_slice(&data[relay_start..relay_start + 4]);
+    let relay_instructions_len = u32::from_le_bytes(relay_len_bytes) as usize;
+
+    let relay_data_start = relay_start + 4;
+    if data.len() < relay_data_start + relay_instructions_len {
+        return Err(ExecutorQuoterError::InvalidInstructionData.into());
+    }
+
+    let relay_instructions = &data[relay_data_start..relay_data_start + relay_instructions_len];
+
+    // Parse relay instructions
+    let (gas_limit, msg_value) = parse_relay_instructions(relay_instructions)?;
+
+    // Calculate quote using U256 math
+    let required_payment = math::estimate_quote(
+        quote_body.base_fee,
+        quote_body.src_price,
+        quote_body.dst_price,
+        quote_body.dst_gas_price,
+        chain_info.gas_price_decimals,
+        chain_info.native_decimals,
+        gas_limit,
+        msg_value,
+    )?;
+
+    // Return data layout (96 bytes, matching EVM return values):
+    // - bytes 0-31: required_payment (U256, big-endian)
+    // - bytes 32-63: payee_address (32 bytes)
+    // - bytes 64-95: quote_body (32 bytes, EQ01 format)
+    let mut return_data = [0u8; 96];
+    return_data[0..32].copy_from_slice(&required_payment.to_be_bytes());
```

Comment:
for svm I'd assume required payment should be u64 since that's what `system_instruction::transfer` takes as an `amount`